services/net/transcription/TranscriptionManager.cs

```diff
@@ -0,0 +1,206 @@
+
+        // Upload the file to the API.
+        if (content != null && !String.IsNullOrWhiteSpace(result.Value.FilePath))
+        {
+            // TODO: Handle different storage locations.
+            // Remote storage locations may not be easily accessible by this service.
+            // TODO: This allows the captured stream to be converted to text - fix.
+            var sourcePath = Path.Join(_options.ClipPath, result.Value.FilePath);
+            if (File.Exists(sourcePath))
+            {
+                var fileBytes = File.ReadAllBytes(sourcePath);
+                var transcript = await GetTranscriptionAsync(fileBytes);
+                content.Transcription = transcript;
+
+                content = await _api.UpdateContentAsync(content);
+                this.Logger.LogDebug("Content Updated.  Content ID: {Id}", content?.Id);
+            }
+            else
+            {
+                this.Logger.LogError("File not found.  Content ID: {Id}, File: {sourcePath}", content.Id, sourcePath);
+            }
+        }
+        else
+        {
+            this.Logger.LogDebug("Content already exists. Content Source: {Source}, UID: {Uid}", content.Source, content.Uid);
+        }
+    }
+
+    /// <summary>
+    /// Stream the audio file to Azure and return the speech to text output.
+    /// </summary>
+    /// <param name="fileBytes"></param>
+    /// <returns></returns>
+    public async Task<string> GetTranscriptionAsync(byte[] fileBytes)
+    {
+        var sem = new Semaphore(0, 1);
+        var sb = new StringBuilder("");
+        var config = SpeechTranslationConfig.FromSubscription(_options.AzureCognitiveServicesKey, _options.AzureRegion);
+        config.SpeechRecognitionLanguage = "en-US";
+
+        AudioStreamFormat audioStreamFormat = AudioStreamFormat.GetCompressedFormat(AudioStreamContainerFormat.MP3);
+        PushAudioInputStream audioStream = PushAudioInputStream.CreatePushStream(audioStreamFormat);
+
+        audioStream.Write(fileBytes);
+
+        AudioConfig audioConfig = AudioConfig.FromStreamInput(audioStream);
+        using (var recognizer = new SpeechRecognizer(config, audioConfig))
+        {
+            recognizer.Recognized += (s, e) =>
+            {
+                var result = e.Result;
+                if (result.Reason == ResultReason.RecognizedSpeech)
+                {
+                    sb.Append(result.Text);
+                }
+            };
+
+            recognizer.Canceled += (s, e) =>
+            {
+                sem.Release();
+            };
+
+            recognizer.SessionStopped += (s, e) =>
+            {
+                sem.Release();
+            };
+
+            // Starts continuous recognition.
+            // Uses StopContinuousRecognitionAsync() to stop recognition.
+            await recognizer.StartContinuousRecognitionAsync().ConfigureAwait(false);
+            audioStream.Close();
```

Comment:
An error will result in the audioStream not being closed.  Which can result in a memory leak.  Possible solution may be a `using` statement, or appropriate `try+catch`

Reply:
Looks like an AudioStream is an IDisposable object. The documentation says that the "using" expression ensures that its corresponding variable is disposed of properly when the method exits. I've used the new syntax which avoids the need for braces, so the two IDisposble variables used in GetTranscriptionAsync() are declared as:

        using var audioStream = PushAudioInputStream.CreatePushStream(audioStreamFormat);
        using var recognizer = new SpeechRecognizer(config, audioConfig);

I've tested this with no issues. Is this what you were looking for?